File: 125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 4/3/2026, 2:02:23 PM
class Solution {
public:
    bool isPalindrome(string s) {
        string newString;

        // Filter out non-alphanumeric characters and convert to lowercase
        for (char c : s) {
            if (isalnum(c)) {
                newString += tolower(c);
            }
        }

        // Check if the filtered string is a palindrome
        int left = 0, right = newString.size() - 1;
        while (left < right) {
            if (newString[left] != newString[right]) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 4/3/2026, 2:02:26 PM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if (s.isalnum()):
            if (s == s[::-1]):
                return True
        s = s.lower()
        s1 = ""
        for i in s:
            if i.isalnum():
                s1 = s1+i

        if (s1 == s1[::-1]):
            return True
        return False